### PR TITLE
Fix warnings about unexpected cfg(feature) values

### PR DESCRIPTION
**Pull request by svix-jplatte**
## Motivation

CI is failing due to a new rustc lint. More about this lint in the [recent blog post](https://blog.rust-lang.org/2024/05/06/check-cfg.html).

## Solution

See individual commits. I chose different solutions for different parts of the codebase where warnings were emitted.

### CODE REVIEW

File: tracing-tower/src/lib.rs [hunk 1]
```diff
@@ -71,33 +71,8 @@ where
     }
 }
 
-#[cfg(feature = "tower-util")]
```
Comment (Xiretza) [inline tracing-tower/src/lib.rs:74]: These could be migrated to `tower_make` instead, see #2971

**Review by davidbarsky — APPROVED** [review]
thanks!

File: tracing-tower/src/lib.rs (hunk 1 above)
Comment (svix-jplatte) [reply tracing-tower/src/lib.rs:74]: Yeah, see the commit description for why I didn't do that.

**Comment by svix-jplatte** [comment]
@davidbarsky any reason this wasn't merged after approval?

**Comment by davidbarsky** [comment]
> @davidbarsky any reason this wasn't merged after approval?

No reason other than "Phabricator taught me muscle memory that the PR author can land PRs/diffs after approval" 😅. Landing this.

**Pull request merged**